args/core/ecs/entity_handle.hpp

```diff
@@ -217,6 +258,17 @@ namespace args::core::ecs
          */
         component_handle_base add_component(id_type componentTypeId) const;
 
+        /**@brief Add component to the entity.
+         * @param componentTypeId Type id of component to add.
+         * @param value Pointer to component_type that has the starting value you require.
+         * @throws args_invalid_entity_error Thrown when handle's registry reference is invalid.
+         * @throws args_entity_not_found_error Thrown when handle's id is invalid.
+         * @throws args_unknown_component_error Thrown when the component type is unknown.
+         * @returns component_handle_base Valid component handle base for the newly created component.
+         * @note component_handle_base needs to be force_cast to component_handle<T> in order to be usable.
+         */
+        component_handle_base add_component(id_type componentTypeId, void* value) const;
+
```

Comment:
component_handle_base should either be a proxy or have a builtin converter to reduce the overhead of casting to the required type